Mathlib/SetTheory/Ordinal/Basic.lean

```diff
@@ -1451,8 +1451,7 @@ theorem type_fin (n : ℕ) : @type (Fin n) (· < ·) _ = n := by simp
 
 end Ordinal
 
-/-! ### Sorted lists -/
-
+@[deprecated (since := "2024-09-20")]
 theorem List.Sorted.lt_ord_of_lt [LinearOrder α] [IsWellOrder α (· < ·)] {l m : List α}
     {o : Ordinal} (hl : l.Sorted (· > ·)) (hm : m.Sorted (· > ·)) (hmltl : m < l)
     (hlt : ∀ i ∈ l, Ordinal.typein (· < ·) i < o) : ∀ i ∈ m, Ordinal.typein (· < ·) i < o := by
```

Comment:
Instead of inlining, could you extract out the lemma that `m ≤ l` implies that `∀ i ∈ m, ∃ j ∈ l, i ≤ j`?